perf(linter/no-unused-vars): use default IgnorePattern when /^_/ is provided as a pattern

DonIsaac opened the pull request:
Use `IgnorePattern::Default` even when `/^_/` is explicitly provided as a regex. This gives better diagnostic messages and lets us use `starts_with('_')` in more cases.

DonIsaac commented:
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/6697?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#6697** <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/6697?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a> 👈
* **#6696** <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/6696?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a>
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @DonIsaac and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="11px" height="11px"/> <b>Graphite</b></a>
  <!-- Current dependencies on/for this PR: -->

camchenry commented on crates/oxc_linter/src/rules/eslint/no_unused_vars/options.rs:
```diff
@@ -657,7 +659,7 @@ mod tests {
         // option object provided, no default varsIgnorePattern
         assert!(rule.vars_ignore_pattern.is_none());
-        assert!(rule.args_ignore_pattern.unwrap().as_str() == "^_");
+        assert!(matches!(rule.args_ignore_pattern, IgnorePattern::Default));
```
Can use is_default in this file instead:
```suggestion
        assert!(rule.args_ignore_pattern.is_default());
```

DonIsaac commented:
### Merge activity

* **Oct 20, 12:24 AM EDT**: The merge label '0-merge' was detected. This PR will be added to the [Graphite merge queue](https://app.graphite.dev/merges?org=oxc-project&repo=oxc) once it meets the requirements.

DonIsaac committed:
perf(linter/no-unused-vars): use default IgnorePattern when /^_/ is provided as a pattern (#6697)

Use `IgnorePattern::Default` even when `/^_/` is explicitly provided as a regex. This gives better diagnostic messages and lets us use `starts_with('_')` in more cases.
